### ISSUE TITLE
Connection to device succeeded but no datapoints found

### ISSUE THREAD

**Issue by ThePapaG** [rospogrigio/localtuya#1730]
## The problem

Trying to add the 2 devices of `FLG0153HA`. Device is not discovered and even getting keys from tuya-cli doesn't add the device.

![image](https://github.com/rospogrigio/localtuya/assets/38738958/cc0d7321-697c-4a70-abe2-a69843b40f59)

[config_entry-localtuya-02211510b02b7a27ffcc3a4deaecf621.json](https://github.com/user-attachments/files/15787240/config_entry-localtuya-02211510b02b7a27ffcc3a4deaecf621.json)

## Environment

- Localtuya version:  
- Home Assistant Core version:  
- [x] Does the device work using the Home Assistant Tuya Cloud component ? 
- [] Does the device work using the Tinytuya (https://github.com/jasonacox/tinytuya) command line tool ? 
- [x] Was the device working with earlier versions of localtuya ? Which one? 
- [x] Are you using the Tuya/SmartLife App in parallel ? 

## Steps to reproduce

1. Add device to smart life
2. setup tuya cloud
3. use tuya-cli to get keys
4. attempt to add device

## Provide Home Assistant traceback/logs

```
This error originated from a custom integration.

Logger: homeassistant
Source: custom_components/localtuya/discovery.py:67
integration: LocalTuya (documentation, issues)
First occurred: 9:31:24 PM (270 occurrences)
Last logged: 9:42:44 PM

Error doing job: Exception in callback _SelectorDatagramTransport._read_ready() (None)
Traceback (most recent call last):
  File "/config/custom_components/localtuya/discovery.py", line 65, in datagram_received
    data = decrypt_udp(data)
           ^^^^^^^^^^^^^^^^^
  File "/config/custom_components/localtuya/discovery.py", line 30, in decrypt_udp
    return _unpad(decryptor.update(message) + decryptor.finalize()).decode()
                                              ^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.12/site-packages/cryptography/hazmat/primitives/ciphers/base.py", line 184, in finalize
    data = self._ctx.finalize()
           ^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.12/site-packages/cryptography/hazmat/backends/openssl/ciphers.py", line 223, in finalize
    raise ValueError(
ValueError: The length of the provided data is not a multiple of the block length.

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/usr/local/lib/python3.12/asyncio/events.py", line 88, in _run
    self._context.run(self._callback, *self._args)
  File "/usr/local/lib/python3.12/asyncio/selector_events.py", line 1238, in _read_ready
    self._protocol.datagram_received(data, addr)
  File "/config/custom_components/localtuya/discovery.py", line 67, in datagram_received
    data = data.decode()
           ^^^^^^^^^^^^^
UnicodeDecodeError: 'utf-8' codec can't decode byte 0xd5 in position 1: invalid continuation byte
```

**Comment by ehran** [comment]
Having similar issues, I tried adding a wifi energy meter, integrated but unable to find DPs, and then another device (water valve) crashed and I am unable to add it back (its getting added but then shows offline whereas in the Tuya app and integration it works well) 
Any issue with localtuya's latest version ?

**Comment by Siffup** [comment]
Same over here with 2 temp/hum devices.

Logger: custom_components.localtuya.pytuya
Source: custom_components/localtuya/pytuya/__init__.py:863


[bf7...afw] Failed to get status:
Traceback (most recent call last):
  File "/usr/local/lib/python3.12/asyncio/tasks.py", line 520, in wait_for
    return await fut
           ^^^^^^^^^
  File "/usr/local/lib/python3.12/asyncio/locks.py", line 386, in acquire
    await fut
asyncio.exceptions.CancelledError

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/config/custom_components/localtuya/pytuya/__init__.py", line 863, in detect_available_dps
    data = await self.status()
           ^^^^^^^^^^^^^^^^^^^
  File "/config/custom_components/localtuya/pytuya/__init__.py", line 795, in status
    status = await self.exchange(DP_QUERY)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/config/custom_components/localtuya/pytuya/__init__.py", line 769, in exchange
    msg = await self.dispatcher.wait_for(seqno, payload.cmd)
          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/config/custom_components/localtuya/pytuya/__init__.py", line 450, in wait_for
    await asyncio.wait_for(self.listeners[seqno].acquire(), timeout=timeout)
  File "/usr/local/lib/python3.12/asyncio/tasks.py", line 519, in wait_for
    async with timeouts.timeout(timeout):
  File "/usr/local/lib/python3.12/asyncio/timeouts.py", line 115, in __aexit__
    raise TimeoutError from exc_val
TimeoutError

**Comment by ehran** [comment]
@neildsb is that a different version of localtuya than the one on HACS ?

**Comment by neildsb** [comment]
> @neildsb is that a different version of localtuya than the one on HACS ?

it's a fork, and works 

see https://github.com/xZetsubou/hass-localtuya/discussions/79

**Comment by ehran** [comment]
thanks, will try :)

**Comment by ehran** [comment]
@neildsb I installed the fork following the instructions, all my devices seem to be recognized so all is well. 
I then proceeded to add the energy meter, I see that there is a new protocol (3.5) available however after trying different options (3.4, 3.5, manual DPs etc..) I am still getting the error message "Connection to device succeeded but no datapoints could be found. Please try set-up again. If problem persists, create a new issue (including debug logs)."

Any hints ? :'( 

Thanks !

**Comment by Pico1965** [comment]
Same here with QS-WIFI-SEC02 switch:


`2024-06-19 16:03:09.544 ERROR (MainThread) [custom_components.localtuya.pytuya] [bf0...ls6] Failed to get status: 
Traceback (most recent call last):
  File "/usr/local/lib/python3.12/asyncio/tasks.py", line 520, in wait_for
    return await fut
           ^^^^^^^^^
  File "/usr/local/lib/python3.12/asyncio/locks.py", line 386, in acquire
    await fut
asyncio.exceptions.CancelledError

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/config/custom_components/localtuya/pytuya/__init__.py", line 863, in detect_available_dps
    data = await self.status()
           ^^^^^^^^^^^^^^^^^^^
  File "/config/custom_components/localtuya/pytuya/__init__.py", line 795, in status
    status = await self.exchange(DP_QUERY)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/config/custom_components/localtuya/pytuya/__init__.py", line 769, in exchange
    msg = await self.dispatcher.wait_for(seqno, payload.cmd)
          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/config/custom_components/localtuya/pytuya/__init__.py", line 450, in wait_for
    await asyncio.wait_for(self.listeners[seqno].acquire(), timeout=timeout)
  File "/usr/local/lib/python3.12/asyncio/tasks.py", line 519, in wait_for
    async with timeouts.timeout(timeout):
  File "/usr/local/lib/python3.12/asyncio/timeouts.py", line 115, in __aexit__
    raise TimeoutError from exc_val
TimeoutError
2024-06-19 16:03:09.582 DEBUG (MainThread) [custom_components.localtuya.config_flow] Initial state update failed (), trying reset command
2024-06-19 16:03:09.583 DEBUG (MainThread) [custom_components.localtuya.config_flow] No DPS able to be detected: object of type 'NoneType' has no len()
2024-06-19 16:03:09.583 DEBUG (MainThread) [custom_components.localtuya.config_flow] Detected DPS: {}
2024-06-19 16:03:13.444 INFO (MainThread) [custom_components.localtuya.common] [bf6...l6l] Trying to connect to 192.168.0.228...
2024-06-19 16:03:13.799 ERROR (MainThread) [custom_components.localtuya.common] [bf6...l6l] Initial state update failed, giving up: DecodeError('could not decrypt data: wrong local_key? (exception: Expecting value: line 1 column 1 (char 0))')
2024-06-19 16:03:13.803 WARNING (MainThread) [custom_components.localtuya.common] [bf6...l6l] Disconnected - waiting for discovery broadcast
2024-06-19 16:03:18.450 INFO (MainThread) [custom_components.localtuya.common] [bf6...l6l] Trying to connect to 192.168.0.228...
2024-06-19 16:03:18.718 ERROR (MainThread) [custom_components.localtuya.common] [bf6...l6l] Initial state update failed, giving up: DecodeError('could not decrypt data: wrong local_key? (exception: Expecting value: line 1 column 1 (char 0))')
2024-06-19 16:03:18.721 WARNING (MainThread) [custom_components.localtuya.common] [bf6...l6l] Disconnected - waiting for discovery broadcast`

Thanks

**Comment by almighty059** [comment]
Same issue.

**Comment by Siffup** [comment]
...and a PTZ camera.

**Comment by Pico1965** [comment]
Sorry... I don't want to seem rude or pretentious.
I am also aware that nothing is due here.
None of the users who reported the problem received a qualified response other than to move to a fork release.

I therefore think it is right to ask whether this app is still maintained by the developer or not.

Thanks for your attention.

**Comment by almighty059** [comment]
I switched to the below version of Local Tuya and everything is now working. It seems to be a more recent and up-to-date maintained version.

https://github.com/xZetsubou/hass-localtuya

**Comment by CodeFatherG** [comment]
> I switched to the below version of Local Tuya and everything is now working. It seems to be a more recent and up-to-date maintained version.
> 
> https://github.com/xZetsubou/hass-localtuya

@rospogrigio are you maintaining this repo. If not then we should look at switching/removing this repo from hacs. The readme should be updated to make it EXTREMELY clear that this is unmaintained.

**Comment by Pico1965** [comment]
> I switched to the below version of Local Tuya and everything is now working. It seems to be a more recent and up-to-date maintained version.
> 
> https://github.com/xZetsubou/hass-localtuya

I've many Tuya device.

Can the two applications coexist on the same system (naturally managing different devices)?

**Comment by rospogrigio** [comment]
> > I switched to the below version of Local Tuya and everything is now working. It seems to be a more recent and up-to-date maintained version.
> > https://github.com/xZetsubou/hass-localtuya
> 
> @rospogrigio are you maintaining this repo. If not then we should look at switching/removing this repo from hacs. The readme should be updated to make it EXTREMELY clear that this is unmaintained.

I am planning to restart maintaing this repo in the near future but I am currently undergoing hard personal issues which don't let me. Perhaps I'd better point it out in the README as you proposed, will do it soon.
Thank you

**Comment by rospogrigio** [comment]
OK, actually I've been trying the above linked repo and it is not working for me (only some devices are), but I don't have even time to check what's going wrong...

**Comment by Pico1965** [comment]
In our area they say: whoever leaves the old road for the new one knows what he leaves but I don't know what he finds"!

@rospogrigio :)

**Comment by CodeFatherG** [comment]
To be transparent, I just decided the official cloud integration was fine. I get delays but it works. Would like to go local but with the bugs in this integration I settled for function. 

No clue if the fork works or how well.

**Comment by Pico1965** [comment]
Upgrading to release 5.2.2 did not bring any benefits...

**Comment by dombrowa** [comment]
Another cause can be that the device's key in the cloud changed (this happened to one of my devices)
"Error when adding in Local Tuya
Connection to device succeeded but no datapoints found, please try again. Create a new issue and include debug logs if problem persists.
"
apparently indicates that the device/endpoint is found (local tuya can find this IP and likely connect to ist servce) but not necessarily query the device because it needs a valid key to authenticate any request.
The error message is not clear IMHO.

**Comment by cfrancisco** [comment]
During the configuration screen, set the Manual DPS's to 0, it worked for me.
